src/utils/fromJS.js

```diff
@@ -7,6 +7,21 @@
  */
 
 const Immutable = require("immutable");
+const isFunction = require("lodash/isFunction");
+
+// hasOwnProperty is defensive because it is possible that the
+// object that we're creating a map for has a `hasOwnProperty` field
+function hasOwnProperty(value, key) {
```

Comment:
this feels weird, do we have a write up on what the issue is? 

Since it solves the issue, and it is used in a controlled way, lets merge it for now -- and investigate later?